platform/ios/src/MGLCompactCalloutView.m

```diff
@@ -8,6 +8,16 @@ @implementation MGLCompactCalloutView
 }
 
 @synthesize representedObject = _representedObject;
+@synthesize anchoredToAnnotation = _anchoredToAnnotation;
+@synthesize dismissesAutomatically = _dismissesAutomatically;
+
+- (instancetype)init {
```

Comment:
`-init` is not the designated initializer for UIView instances; that’d be either `-initWithFrame:` or `-initWithCoder:`. This class’s `+platformCalloutView` implementation happens to call `-init`, but this is a bit of an antipattern.

Instead of maintaining ivars for read-only variables, implement `-anchoredToAnnotation` and `-dismissesAutomatically` to return `YES` and `NO`, respectively.